Inc/HALAL/Models/MDMA/MDMA.hpp

```diff
@@ -0,0 +1,317 @@
+	 * @return the id that represents the MDMA channel with its designated buffer inside this utility class, used in all its functions.
+	 */
+
+    static uint8_t inscribe(uint8_t* data_buffer, uint8_t* destination_address=nullptr);
+
+    /**
+	 * @brief A method to create a linked list from a tuple of pointers that come from packets, this linked list are global for all instances.
+
+	 * @param MDMA_id	the id that represents the MDMA channel instance
+     * @param values  the tuple of pointers that will be added to the linked list
+     * 
+	 * @return the id that represents the linked list associated to the packet.
+	 */
+    template<typename... pointers>
+    static uint8_t add_packet(const uint8_t MDMA_id,const std::tuple<pointers...>& values);
+
+    /**
+     * @brief A method to merge multiple packets into a single linked list.
+     * 
+     * @param MDMA_id The ID of the MDMA channel instance.
+     * @param base_packet_id The ID of the base packet to which other packets will be merged.
+     * @param packets_id The IDs of the packets to be merged into the base packet.
+     * 
+     * @return The ID of the merged linked list.
+     */
+    template<typename... PacketIds>
+    static uint8_t merge_packets(const uint8_t MDMA_id,const uint8_t base_packet_id, const PacketIds... packets_id);
+
+    /**
+     * @brief A method to start a transfer from source to destination using MDMA linked list
+     *  
+     * @param MDMA_id The ID of the MDMA channel instance.
+     * @param packet_id The ID of the linked list packet to be transferred.
+     * @param destination_address The destination address for the transfer. If nullptr, the default destination associated with the instance will be used.
+     * @param promise An optional promise to be fulfilled upon transfer completion.
+     */
+    static void transfer_data(const uint8_t MDMA_id,uint8_t* source_address, const uint32_t data_length,uint8_t* destination_address=nullptr,Promise* promise=nullptr);
+
+    /**
+     * @brief A method to transfer a packet using MDMA linked list
+     *  
+     * @param MDMA_id The ID of the MDMA channel instance.
+     * @param packet_id The ID of the linked list packet to be transferred.
+     * @param destination_address The destination address for the transfer. If nullptr, the default destination associated with the instance will be used.
+     * @param promise An optional promise to be fulfilled upon transfer completion.
+     */
+    static void transfer_packet(const uint8_t MDMA_id, const uint8_t packet_id, uint8_t* destination_address=nullptr, Promise* promise=nullptr);
+
+};
+
+template<typename... pointers>
+inline uint8_t MDMA::add_packet(const uint8_t MDMA_id, const std::tuple<pointers...>& values)
+{
+    Instance& instance = get_instance(MDMA_id);
+    uint32_t offset{0};
+    HAL_StatusTypeDef status;
+    const uint8_t current_packet_id = number_of_packets++;
+
+    std::vector<MDMA_LinkNodeTypeDef>& nodes = linked_lists[current_packet_id];
```

Comment:
Again, avoid using STL container that make use of heap, use array with max size, as you receive a tuple of pointers you could even get the size from there